SCClassLibrary/Common/GUI/PlusGUI/Math/PlotView.sc

```diff
@@ -65,75 +66,152 @@ Plot {
 			backgroundColor = ~background;
 			this.gridLineSmoothing = ~gridLineSmoothing;
 			this.gridLinePattern = ~gridLinePattern !? {~gridLinePattern.as(FloatArray)};
-			labelX = ~labelX;
+			labelX = ~labelX; // axis label
 			labelY = ~labelY;
 		};
 		resolution = plotter.resolution;
 	}
 
 	bounds_ { |viewRect|
-		var gridRect;
-		var xtkLabels, ytkLabels, ytkLabelSize, xtkLabelSize, xLabelGridOffset, yLabelGridOffset;
-		var xLabelSize, yLabelSize, maxWidthWithLabel, maxHeightWithLabel;
-		var hshift, vshift, hinset, vinset, lmargin, rmargin, totalVSpace;
-		var minGridMargin = 4;
+		var gridRect, sizeAllowsXLabels, sizeAllowsYLabels;
 
-		bounds = viewRect;
 		valueCache = nil;
+		bounds = viewRect;
+
+		#gridRect, sizeAllowsXLabels, sizeAllowsYLabels = this.prCalcLabelSpace(viewRect);
+
+		drawGrid.x.showLabels = sizeAllowsXLabels and: { showTickLabelsX };
+		drawGrid.y.showLabels = sizeAllowsYLabels and: { showTickLabelsY };
+		drawGrid.bounds = gridRect;
+		plotBounds = gridRect;
+	}
 
-		// find the size of the largest tick labels
-		#xtkLabelSize, xtkLabels = this.prGetTickLabelSize(drawGrid.x);
-		#ytkLabelSize, ytkLabels = this.prGetTickLabelSize(drawGrid.y);
-		xLabelSize = labelX.notNil.if({ labelX.bounds(font) }, { Rect(0,0,0,0) });
-		yLabelSize = labelY.notNil.if({ labelY.bounds(font) }, { Rect(0,0,0,0) });
-
-		// zeroing out drawGrids' labelOffsets disables labels, see DrawGridX/Y:commands
-		xLabelGridOffset = yLabelGridOffset = Point(0, 0);
-
-		// labels disappear below these size thresholds
-		maxWidthWithLabel = ytkLabelSize.width * 4;
-		maxHeightWithLabel = xtkLabelSize.height * 4;
-
-		hshift = vshift = hinset = vinset = 0;
-
-		// x axis label spacing
-		if( viewRect.height >= maxHeightWithLabel
-			and: { viewRect.width >= (xtkLabelSize.width*3)
-				and: { xtkLabels.notNil }
-		}) {
-			totalVSpace = txtPad + xtkLabelSize.height + txtPad + xLabelSize.height + txtPad;
-			hinset = xtkLabelSize.width/2 + txtPad;
-			vinset = totalVSpace * (2/3);     // inset more on sides with labels
-			vshift = totalVSpace * (1/3).neg;
-			xLabelGridOffset = Point(xtkLabelSize.width, xtkLabelSize.height);
+	// Calculate grid offset to accomodate tick and axis labels,
+	// return modified gridRect and bools allowing X and Y labels.
+	// Currently only supports tick labels that don't extend past
+	// the side opposite their respective axis.
+	// TODO: this method could be refactored to use the showTickLabelsX/Y
```

Comment:
isn't that done? Line 111 does the skip.

Reply:
Line 111 is a shortcut in case _both_ `showTickTickLabels` vars are false. This comment points to refactoring further when only one is disabled, then there can likely be some calculation skipped.